src/main/java/org/llorllale/cactoos/matchers/Mismatches.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) for portions of project cactoos-matchers are held by
+ * Yegor Bugayenko, 2017-2018, as part of project cactoos.
+ * All other copyright for project cactoos-matchers are held by
+ * George Aristy, 2018.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.llorllale.cactoos.matchers;
+
+import org.cactoos.Text;
+import org.cactoos.text.Joined;
+import org.cactoos.text.TextOf;
+import org.cactoos.text.UncheckedText;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+/**
+ * Matcher to test {@link org.hamcrest.Matcher} objects.
+ *
+ * <p>Here is an example:</p>
+ * <pre>{@code
+ *  new Assertion<>(
+ *      "Can't describe a mismatch",
+ *      new HasLines("Tom", "Mike"),
+ *      new Mismatches<>(
+ *          String.format("Tom%nJohn%n"),
+ *          "Lines are <[Tom, Mike]>",
+ *          "<[Tom, John]>"
+ *      )
+ *  ).affirm();
+ *  }</pre>
+ *
+ * @param <X> Type of item.
+ * @since 1.0.0
+ * @todo #106:30min Convert all the Matcher tests to use Mismatches
+ *  instead of checking directly the output of mismatch methods or
+ *  the exception of Assertion.
+ * @todo #106:30min Add extra tests for this class to validate all
+ *  the different constructors and ensure they are coherent with how
+ *  Assertion is working and throwing errors.
+ * @checkstyle ProtectedMethodInFinalClassCheck (200 lines)
+ */
+public final class Mismatches<X> extends TypeSafeDiagnosingMatcher<Matcher<X>> {
```

Comment:
@victornoel since there is a task for refactoring all matchers to extend `MatcherEnvelope`, isn't it better to do so from the start here?

Reply:
@scristalli I could, but this would make the `Mismatches` class very hard to read I believe. When there will be extra code for solving the `@todo`s of the class, it will be even harder. `MatcherEnvelope` is nice when the 3 implementations of the methods are very simple.

Reply:
@scristalli see also #135

Reply:
@victornoel I see your point, and read the new issue you opened. I still think this creates somewhat of a special case for #129 (and similar) though, should be either clarified or avoided. I think `ARC` has the final decision here.